mera/mera_pymer4.py

```diff
@@ -159,6 +172,44 @@ def run_mera(
         else:
             print("WARNING: No data for IM, skipping...")
 
+        if verbose_warnings:
+            # Store the warning strings to print at the end
+            warning_str_per_im[cur_im]["stat_id"].extend(
+                utils.generate_insufficient_records_warning_str(
+                    cur_residual_df, "stat_id", min_num_records_per_site
+                )
+            )
+
+            warning_str_per_im[cur_im]["event_id"].extend(
+                utils.generate_insufficient_records_warning_str(
+                    cur_residual_df, "event_id", min_num_records_per_event
+                )
+            )
+
+    if verbose_warnings:
+        AnyWarnings = False
+
+        # Check if there are any warnings
+        for im in ims:
+            if (
+                len(warning_str_per_im[im]["stat_id"]) > 0
+                or len(warning_str_per_im[im]["event_id"]) > 0
+            ):
+                AnyWarnings = True
+                break
+
+        if AnyWarnings:
+            # Print the warnings about records with insufficient number of sites or events
+            print("Warnings about records with insufficient number of sites or events:")
+            for im in ims:
+                print(f"IM: {im}")
+                print("For stat_id:")
+                for x in range(len(warning_str_per_im[im]["stat_id"])):
```

Comment:
any reason why you're iterating over a range instead of using something like `iterrows`?